src/components/TreeGrid.ts

```diff
@@ -46,13 +46,67 @@ const wrapContent =
     state: 'closed',
   });
 
+const hierarchyMapToRows = <T extends RowContents>(hierarchyMap: Map<number, T[]>, domainOptionRoot: T): Row<T>[] => {
+  const traverseHierarchy = (parent: T, depth: number, rows: Row<T>[]): Row<T>[] => {
+    const updatedRows: Row<T>[] = [
+      ...rows,
+      {
+        contents: parent,
+        depth,
+        isFetched: true,
+        state: 'open',
+      },
+    ];
+
+    // fetch all children of parent
+    const children: T[] | undefined = hierarchyMap.get(parent.id);
+    // if parent doesn't have children, return updatedRows
+    if (!children) {
+      return updatedRows;
+    }
```

Comment:
I think you could use `||` here to provide an empty list for the `forEach()` instead of returning early.
```suggestion
    const children: T[] = hierarchyMap.get(parent.id) || [];
```

Reply:
This is a great suggestion!!!! Thank you!